model/task/task.go

```diff
@@ -2036,12 +2028,25 @@ func (t *Task) GetDisplayStatus() string {
 		}
 		return evergreen.TaskSystemFailed
 	}
-	if t.Details.Type == evergreen.CommandTypeSetup {
-		return evergreen.TaskSetupFailed
-	}
 	if t.Details.TimedOut {
 		return evergreen.TaskTimedOut
 	}
+	if t.Status == evergreen.TaskUndispatched {
+		if !t.Activated {
+			return evergreen.TaskUnscheduled
+		}
+		if t.Blocked() {
+			return evergreen.TaskStatusBlocked
+		}
+		return evergreen.TaskWillRun
+	}
+	if !t.IsFinished() {
+		return t.Status
+	}
+	annotation, err := annotations.FindOneByTaskIdAndExecution(t.Id, t.Execution)
+	if err == nil && annotation != nil {
+		return evergreen.TaskKnownIssue
```

Comment:
I think it should only be a known issue if the annotation known issues list isn't empty